recipes/grpc/all/conanfile.py

```diff
@@ -54,11 +54,11 @@ def _build_subfolder(self):
 
     def requirements(self):
         self.requires('zlib/1.2.11')
-        self.requires('openssl/1.1.1k')
+        self.requires('openssl/1.1.1l')
         self.requires('protobuf/3.17.1')
         self.requires('c-ares/1.17.1')
-        self.requires('abseil/20210324.1')
-        self.requires('re2/20210401')
+        self.requires('abseil/20210324.2')
+        self.requires('re2/20210601')
 
     def build_requirements(self):
         self.build_requires('protobuf/3.17.1')
```

Comment:
IMHO, if it works, we should try to avoid adding conditionals to recipes. We need to think in the context of two profiles and _fix_ the recipe if needed for consumers that use only one profile.

Also, in this concrete recipe, I think we won't find the problem you are referring to because the requirement and the build-requirement are exactly the same and there will never be a conflict when using one profile. It is different when using a build-requirement like `cmake` that also requires some libraries that appear in the _host_ context, in that scenario it requires to change `cmake` first and then the recipe that uses it.

Reply:
> the build-requirement are exactly the same and there will never be a conflict when using one profile

It's not true. requirements can be overridden, build_requirements can't, and in case of 1 profile it's a conflict leading to an error.